File: examples/python/good/srp.py
class Reader:
    def read(self):
        # reading data...
        print("read data...")


class Parser:
    def parse(self, data):
        # parsing data...
        print("parsing data...")


class Writer:
    def save(self, data):
        # saving data...
        print("save data...")


reader = Reader()
parser = Praser()
writer = Writer()

data = reader.read()
parsed_data = parser.parse(data)
writer.save(parsed_data)

File: examples/typescript/good/srp.ts
class Reader {
    read(): object[] {
        console.log("Reading data...")
        return [{"a": 1, "b": 2}]
    }
}

class Parser {
    parse(data: object[]): object[] {
        // Using data...
        console.log("Parsing data...")
        return data
    }
}

class Saver {
    save(data: object[]): void {
        console.log(`Saving data... ${data}`)
    }
}

(function () {
    const reader = new Reader()
    const parser = new Parser()
    const saver = new Saver()

    let data = reader.read()
    data = parser.parse(data)
    saver.save(data)
 })();